oras/provider.py

```diff
@@ -348,7 +353,7 @@ def get_blob(
         """
         method = "GET" if not head else "HEAD"
         blob_url = f"{self.prefix}://{container.get_blob_url(digest)}"  # type: ignore
-        return self.do_request(blob_url, method, headers=self.headers, stream=stream)
+        return self.do_request(blob_url, method, headers=self.headers, stream=stream, insecure=self._insecure )
```

Comment:
Thank you for opening this! 

Perhaps we don't need to pass self._insecure into a function that already is on the class - we could just use it in the function without doing that?

Reply:
You are right. I made a new commit where the function do_request gets the insecure parameter from the class